src/mlpack/methods/kernel_pca/kernel_rules/naive_method.hpp

```diff
@@ -47,20 +47,24 @@ class NaiveKernelRule
   // Note that we only need to calculate the upper triangular part of the
   // kernel matrix, since it is symmetric. This helps minimize the number of
   // kernel evaluations.
-  for (size_t i = 0; i < data.n_cols; ++i)
+  #pragma omp parallel for
+  for (omp_size_t i = 0; i < data.n_cols; ++i)
   {
     for (size_t j = i; j < data.n_cols; ++j)
     {
       // Evaluate the kernel on these two points.
       kernelMatrix(i, j) = kernel.Evaluate(data.unsafe_col(i),
                                            data.unsafe_col(j));
+      kernelMatrix(j-i,i) = kernelMatrix(i, j);
     }
   }
 
   // Copy to the lower triangular part of the matrix.
+  /*
   for (size_t i = 1; i < data.n_cols; ++i)
     for (size_t j = 0; j < i; ++j)
       kernelMatrix(i, j) = kernelMatrix(j, i);
+  */
```

Comment:
If you're removing code, remove it :) Be bold! Git histories hold the past.

Reply:
Sure :)